src/components/basic_table/basic_table.test.tsx

```diff
@@ -391,24 +391,74 @@ describe('EuiBasicTable', () => {
     expect(container.querySelector('[aria-sort]')).toBeTruthy();
   });
 
-  test('with initial selection', () => {
-    const props: EuiBasicTableProps<BasicItem> = {
-      items: basicItems,
-      columns: basicColumns,
-      itemId: 'id',
-      selection: {
-        onSelectionChange: () => {},
-        initialSelected: [basicItems[0]],
-      },
-    };
-    const { getByTestSubject } = render(<EuiBasicTable {...props} />);
+  describe('selection', () => {
+    const getCheckboxAt = (index: number) =>
+      screen.getByTestSubject(`checkboxSelectRow-${index}`) as HTMLInputElement;
 
-    expect(
-      (getByTestSubject('checkboxSelectRow-1') as HTMLInputElement).checked
-    ).toBeTruthy();
-    expect(
-      (getByTestSubject('checkboxSelectRow-2') as HTMLInputElement).checked
-    ).toBeFalsy();
+    test('initialSelected', () => {
+      const props: EuiBasicTableProps<BasicItem> = {
+        items: basicItems,
+        columns: basicColumns,
+        itemId: 'id',
+        selection: {
+          onSelectionChange: () => {},
+          initialSelected: [basicItems[0]],
+        },
+      };
+      render(<EuiBasicTable {...props} />);
+
+      expect(getCheckboxAt(1).checked).toBeTruthy();
+      expect(getCheckboxAt(2).checked).toBeFalsy();
+    });
+
+    test('selected', () => {
+      const props: EuiBasicTableProps<BasicItem> = {
+        items: basicItems,
+        columns: basicColumns,
+        itemId: 'id',
+        selection: {
+          onSelectionChange: () => {},
+          selected: [basicItems[1]],
+        },
+      };
+      render(<EuiBasicTable {...props} />);
+
+      expect(getCheckboxAt(1).checked).toBeFalsy();
+      expect(getCheckboxAt(2).checked).toBeTruthy();
+    });
+
+    it('ignores initialSelected if selected is passed', () => {
+      const props: EuiBasicTableProps<BasicItem> = {
+        items: basicItems,
+        columns: basicColumns,
+        itemId: 'id',
+        selection: {
+          onSelectionChange: () => {},
+          selected: [],
```

Comment:
derp, I forgot to change this prop
```suggestion
          initialSelected: [basicItems[0]],
          selected: [],
```